File: test/test_custom_script_get_variable_priority6.py
#!/usr/local/bin/python
import coverage_utils
coverage_utils.cov_start()

import time
from std_msgs.msg import Bool
from std_msgs.msg import Int32
import sys
import rospkg
import rospy

rospack = rospkg.RosPack()
mission_control_path = rospack.get_path('mission_control')

sys.path.append("%s/src" % mission_control_path)

import mission_control_utils
from mission_control_utils import Constants

mission_control_utils.ros_init("custom_script_priority6_get")

answer_pub = rospy.Publisher("/mission_control/test/mission_control_utils/answer_test_counter6", Int32, queue_size=Constants.QUEUE_SIZE)
expired_answer_pub = rospy.Publisher("/mission_control/test/mission_control_utils/expired_answer_test_counter6", Int32, queue_size=Constants.QUEUE_SIZE)
start_setting_pub = rospy.Publisher("/mission_control/test/mission_control_utils/start_setting", Bool, queue_size=Constants.QUEUE_SIZE)

time.sleep(2)

start_setting_pub.publish(True)

answer = mission_control_utils.get_var("test_counter6", -1)

answer_pub.publish(int(answer))

time.sleep(10) #Expire var

expired_answer = mission_control_utils.get_var("test_counter6", -1)

expired_answer_pub.publish(expired_answer)

coverage_utils.cov_stop()

File: test/test_custom_script_get_variable_priority6.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <mission_control/mission_control_utils.h>
#include <std_msgs/Bool.h>
#include <std_msgs/Int32.h>

int main (int argc, char* argv[])
{
  ros_init("custom_script_priority6_set", argc, argv);

  ros::NodeHandle nh = ros::NodeHandle();

  ros::Publisher start_setting_pub = nh.advertise<std_msgs::Bool>("/mission_control/test/mission_control_utils/start_setting", QUEUE_SIZE);
  ros::Publisher answer_pub = nh.advertise<std_msgs::Int32>("/mission_control/test/mission_control_utils/answer_test_counter6", QUEUE_SIZE);
  ros::Publisher expired_answer_pub = nh.advertise<std_msgs::Int32>("/mission_control/test/mission_control_utils/expired_answer_test_counter6", QUEUE_SIZE);

  sleep(2);

  std_msgs::Bool msg;
  msg.data = true;
  start_setting_pub.publish(msg);

  ros::spinOnce();

  sleep(2);

  std::string answer = get_var("test_counter6", "-1");

  std_msgs::Int32 answer_msg;
  answer_msg.data = atoi(answer.c_str());
  answer_pub.publish(answer_msg);

  ros::spinOnce();

  sleep(10); //Expire var

  std::string expired_answer = get_var("test_counter6", "-1");

  std_msgs::Int32 expired_answer_msg;
  expired_answer_msg.data = atoi(expired_answer.c_str());
  expired_answer_pub.publish(expired_answer_msg);

  ros::spinOnce();

  sleep(2); //Wait for the message to reach the end

  return 0;
}
